packages/cli/src/commands/package-dynamic-plugins/command.ts

```diff
@@ -13,22 +13,61 @@ import { paths } from '../../lib/paths';
 import { Task } from '../../lib/tasks';
 
 export async function command(opts: OptionValues): Promise<void> {
-  const { exportTo, forceExport, preserveTempDir, tag, useDocker, platform } =
-    opts;
+  const {
+    exportTo,
+    forceExport,
+    preserveTempDir,
+    tag,
+    useDocker,
+    containerTool = 'podman',
+    platform,
+  } = opts;
   if (!exportTo && !tag) {
     Task.error(
       `Neither ${chalk.white('--export-to')} or ${chalk.white('--tag')} was specified, either specify ${chalk.white('--export-to')} to export plugins to a directory or ${chalk.white('--tag')} to export plugins to a container image`,
     );
     return;
   }
-  const containerTool = useDocker ? 'docker' : 'podman';
+
   // check if the container tool is available, skip if just exporting the plugins to a directory
+  let containerToolCmd = containerTool;
+  let containerBuildCommand = 'build';
+
   if (!exportTo) {
+    let _containerTool = containerTool;
+
+    // Check if useDocker is set to true and set the container tool to docker
+    if (useDocker) {
+      Task.log(
+        `The ${chalk.white('--use-docker')} flag is deprecated, use ${chalk.white(
+          '--container-tool',
+        )} instead`,
+      );
+      Task.log(
+        `Setting ${chalk.white('--container-tool')} to docker as ${chalk.white(
+          '--use-docker',
+        )} was specified`,
+      );
+      _containerTool = 'docker';
+    }
+
+    // Validate containerTool value
+    const allowedTools = ['docker', 'podman', 'buildah'];
+    if (!allowedTools.includes(_containerTool)) {
+      Task.error(
+        `Invalid value for --container-tool: ${_containerTool}. Allowed values are: ${allowedTools.join(', ')}`,
+      );
+      return;
+    }
+
+    // Determine the build command and version check command
+    containerBuildCommand = _containerTool === 'buildah' ? 'bud' : 'build';
```

Comment:
Why do we need to use `bud` command in `buildah`, isn't this just a shortcut for `build`?

Reply:
Indeed. Looks like I was looking at a previous version of buildah documentation where this was build using docker file. I'll update the PR